Linking error when using JOYSTICK_DRIVER = digital

Trying to compile qmk with only digital joystick enabled doesn't seem to work. `analogReadPin` is defined in `analog.h`, which I don't need otherwise. Is this a bug or am I missing something?

```
Linking: .build/gmmk_pro_rev1_iso_greaka.elf                                                        [ERRORS]
 |
 | /usr/lib/gcc/arm-none-eabi/12.1.0/../../../../arm-none-eabi/bin/ld: .build/obj_gmmk_pro_rev1_iso_greaka/quantum/process_keycode/process_joystick.o: in function `process_joystick_analogread_quantum':
 | /git/qmk_firmware/quantum/process_keycode/process_joystick.c:110: undefined reference to `analogReadPin'
 | collect2: error: ld returned 1 exit status
 |
make[1]: *** [builddefs/common_rules.mk:364: .build/gmmk_pro_rev1_iso_greaka.elf] Error 1
Make finished with errors
make: *** [Makefile:413: gmmk/pro/rev1/iso:greaka:flash] Error 1
```

--- a/quantum/process_keycode/process_joystick.c
+++ b/quantum/process_keycode/process_joystick.c
@@ -106,7 +106,7 @@ bool process_joystick_analogread_quantum() {
 
         wait_us(10);
 
-#    if defined(__AVR__) || defined(PROTOCOL_CHIBIOS)
+#    if defined(ANALOG_JOYSTICK_ENABLE) && (defined(__AVR__) || defined(PROTOCOL_CHIBIOS))
         int16_t axis_val = analogReadPin(joystick_axes[axis_index].input_pin);
 #    else
         // default to resting position

